src/middlewared/middlewared/plugins/reporting/netdata_configure.py

```diff
@@ -24,17 +25,21 @@ def netdata_state_location(self):
 
     @private
     def post_dataset_mount_action(self):
-        if os.path.exists(get_netdata_state_path()):
+        netdata_state_path = get_netdata_state_path()
+        if os.path.exists(netdata_state_path):
             return
 
-        cp = subprocess.run(
-            ['cp', '-a', '/var/lib/netdata', get_netdata_state_path()], check=False, capture_output=True,
-        )
-        if cp.returncode != 0:
-            self.logger.error('Failed to copy netdata state over from /var/lib/netdata: %r', cp.stderr.decode())
-            # We want to make sure this path exists always regardless of an error so that
-            # at least netdata can start itself gracefully
-            os.makedirs(get_netdata_state_path(), exist_ok=True)
+        # We want to make sure this path exists always regardless of an error so that
+        # at least netdata can start itself gracefully
+        # Also copytree will not apply correct perms on parent dir, so we will apply them as well
+        os.makedirs(netdata_state_path, exist_ok=True)
+        os.chown(netdata_state_path, uid=999, gid=997)
+        os.chmod(netdata_state_path, mode=0o755)
+
+        try:
+            copytree('/var/lib/netdata', netdata_state_path, config=CopyTreeConfig(traverse=True))
```

Comment:
Do we need traverse here? Do we expect to have other filesystems mounted in /var/lib/netdata?

Reply:
I think we should be able to get along without it, testing that bit

Reply:
Yes works as expected without traverse